httpconnect/httpconnect.go

```diff
@@ -147,28 +151,37 @@ func (f *HTTPConnectHandler) intercept(w http.ResponseWriter, req *http.Request)
 			}
 		}
 	}
-	var closeOnce sync.Once
-	go func() {
+
+	readErrCh := make(chan error)
+	op.Go(func() {
 		buf := buffers.Get()
 		defer buffers.Put(buf)
-		if _, err := io.CopyBuffer(connOut, clientConn, buf); err != nil {
-			log.Debug(err)
+		_, readErr := io.CopyBuffer(connOut, clientConn, buf)
+		if readErr != nil {
+			log.Debug(readErr)
 		}
-		closeOnce.Do(closeConns)
-
-	}()
+		readErrCh <- readErr
+	})
 	buf := buffers.Get()
 	defer buffers.Put(buf)
-	if _, err := io.CopyBuffer(clientConn, connOut, buf); err != nil {
-		log.Debug(err)
```

Comment:
With this change, we will wait until both finish, as seen below. This means that if we had an error in one direction, we still should wait for the other to complete for a clean finalization. Is that what you are saying?